### ISSUE TITLE
Poll for new articles

### ISSUE THREAD

**Issue by samibbb** [jameshurst/HNBar#3]
nice to have: Growl integration when a new article becomes top post

**Comment by jameshurst** [comment]
The issue I can see with showing a notification when an article enters the top `n` posts is that if it briefly exists those top `n` posts and reenters the top `n` posts then you'll see the notification for that post twice.

Simply storing a set of articles that have entered the top `n` posts would gradually increase the memory consumption of the app.

**Comment by samibbb** [comment]
good point. I'd still be interested in getting notified when certain types of threads are posted and make it to the front page. 

Some examples of thread types I tend to seek out: Ask HN, Show HN, Who is hiring for that month, Any thread with Objective-C, Cocoa, Swift, Uber in the title etc. 

You could register a set of keywords and only get growl notified when a thread title contains a word in that set

**Comment by jameshurst** [comment]
I believe that would still have the same issue of tracking already notified posts.

**Comment by samibbb** [comment]
Avoid duplicate notifications by tracking them by the website url and not storing them in memory but in NSUserDefaults. flush it periodically if necessary. 

I'll send you a pull request and have you leave Instruments running for a day or two.

**Comment by jameshurst** [comment]
I don't think I'll need a pull request for that. I've got the notifications already implemented on a local branch. All that's left is making sure a notification for one article isn't shown more than once.

**Comment by samibbb** [comment]
Thanks look forward to seeing it

**Comment by jameshurst** [comment]
I decided to go with Core Data to maintain the list of already presented notifications. I've put the notifications stuff on a branch called `notifications` for now. Try it out and let me know how it works.
